src/autocomplete.js

```diff
@@ -666,8 +666,10 @@ class CompletionProvider {
     
         var matches = [];
         this.completers = editor.completers;
-        var total = editor.completers.length;
-        editor.completers.forEach(function(completer, i) {
+        var matchingCompleters = util.getCompletersMatchingTriggerCharacter(editor);
```

Comment:
What happens here if there is a completer with the trigger character `a`, would only the completer with the trigger character get triggered or would all completers still get triggered (as they should given that the user is typing a normal letter)?

Reply:
good question. the autocomplete is triggered when there is a trigger character matching at least one completer or there is a text prefix with non-zero length. string `a` would fall into prefix even if it's a trigger character, but we don't check for it here. I will update implementation to use all completers if there is a prefix.